lib/pengine/rules.c

```diff
@@ -443,74 +443,106 @@ phase_of_the_moon(crm_time_t * now)
     return ((((((diy + epact) * 6) + 11) % 177) / 22) & 7);
 }
 
-#define cron_check(xml_field, time_field)				\
-    value = crm_element_value(cron_spec, xml_field);			\
-    if(value != NULL) {							\
-	gboolean pass = TRUE;						\
-	int rc = pcmk__parse_ll_range(value, &value_low, &value_high);  \
-        if (rc == pcmk_rc_unknown_format) {                             \
-            return FALSE;                                               \
-        } else if (value_low == value_high) {                           \
-            if (value_low != time_field) {                              \
-                pass = FALSE;                                           \
-            }                                                           \
-        } else if (value_low == -1 && value_high != -1) {               \
-            if (time_field > value_high) {                              \
-                pass = FALSE;                                           \
-            }                                                           \
-        } else if (value_low != -1 && value_high == -1) {               \
-            if (value_low > time_field) {                               \
-                pass = FALSE;                                           \
-            }                                                           \
-	} else {                                                        \
-            if (value_low > time_field || value_high < time_field) {    \
-                pass = FALSE;                                           \
-            }                                                           \
-        }                                                               \
-	if(pass == FALSE) {						\
-	    crm_debug("Condition '%s' in %s: failed", value, xml_field); \
-	    return pass;						\
-	}								\
-	crm_debug("Condition '%s' in %s: passed", value, xml_field);	\
+static pe_eval_date_result_t
+check_one(xmlNode *cron_spec, const char *xml_field, uint32_t time_field) {
+    pe_eval_date_result_t rc = pe_date_result_undetermined;
+    const char *value = crm_element_value(cron_spec, xml_field);
+    long long low, high;
+
+    if (value == NULL) {
+        /* Return pe_date_result_undetermined if the field is missing. */
+        goto bail;
     }
 
-gboolean
-pe_cron_range_satisfied(crm_time_t * now, xmlNode * cron_spec)
-{
-    const char *value = NULL;
+    if (pcmk__parse_ll_range(value, &low, &high) == pcmk_rc_unknown_format) {
+       goto bail;
+    } else if (low == -1 && high == -1) {
+        /* A single number was given, not a range. */
+       low = crm_parse_ll(value, NULL);
```

Comment:
Does this need to be rebased on current master? We had reworked this section ...

Reply:
Yeah, good catch.  And there was another bug in this area too so it's a good thing I had to take a second look.